test/test_main.cpp

```diff
@@ -0,0 +1,55 @@
+#define CATCH_CONFIG_MAIN // This tells Catch to provide a main() - only do this in one cpp file
+#include "catch.hpp"
+
+class ObstacleDetection {
```

Comment:
You should test the code you have written by including them, not by pasting the code.